ql/instruments/forwardrateagreement.cpp

```diff
@@ -61,8 +61,72 @@ namespace QuantLib {
     : ForwardRateAgreement(valueDate, index->maturityDate(valueDate), type, strikeForwardRate,
                            notionalAmount, index, std::move(discountCurve), true) {}
 
+    ForwardRateAgreement::ForwardRateAgreement(Position::Type type,
```

Comment:
Type should come after value date for consistency with other constructors. I think Luigi has already suggested modifying the existing constructor rather than adding this one, which would fix this.

Reply:
Yep. Will update this.